docker/pypi/wmagent/bin/manage-common.sh

```diff
@@ -0,0 +1,423 @@
+            else
+                echo "$FUNCNAME: WARNING: Database recorded and current agent's init parameters do NOT match."
+                return $(false)
+            fi
+            ;;
+        *)
+            echo "$FUNCNAME: ERROR: Unknown or not set Agent Flavor"
+            return $(false)
+            ;;
+    esac
+}
+
+_sql_db_isclean(){
+    # Auxiliary function to check if a given database is empty
+    # :param $1: The database name to be checked (it will be ignored for Oracle)
+    echo "$FUNCNAME: Checking if the current SQL Database is clean and empty."
+    local wmaDBName=${1:-$wmaDBName}
+    case $AGENT_FLAVOR in
+        'oracle')
+            echo "Not implemented"
+            ;;
+        'mysql')
+            local sqlCmd="SELECT SCHEMA_NAME   FROM INFORMATION_SCHEMA.SCHEMATA  WHERE SCHEMA_NAME = '$wmaDBName'"
+            local dbExists=$(_exec_mysql "$sqlCmd")
+            [[ -n $dbExists ]] || { echo "$FUNCNAME: WARNING: Database $wmaDBName does not exist. 'manage init-agent' should be executed before starting the agent."; return $(true) ;}
+
+            local sqlCmd="SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = 'wmagent';"
+            local numTables=$(_exec_mysql "$sqlCmd" $wmaDBName)
+            [[ $numTables -eq 0 ]] || { echo "$FUNCNAME: WARNING: numTables=$numTables"; return $(false) ;}
+            ;;
+        *)
+            echo "$FUNCNAME: ERROR: Unknown or not set Agent Flavor"
+            return $(false)
+            ;;
+    esac
+}
+
+_sql_write_agentid(){
+    # Auxiliary function to write the current agent build id into the sql database
+    echo "$FUNCNAME: Preserving the current WMA_BUILD_ID and HostName at database: $wmaDBName."
+    case $AGENT_FLAVOR in
+        'oracle')
+            echo "Not implemented"
+            ;;
+        'mysql')
+            local sqlCmd=""
+
+            echo "$FUNCNAME: Creating wma_init table at database: $wmaDBName"
+            sqlCmd="create table wma_init(initparam varchar(100) not null, initvalue varchar(100));"
+            _exec_mysql "$sqlCmd" $wmaDBName || return
+
+            echo "$FUNCNAME: Inserting current Agent's build id and hostname at database: $wmaDBName"
+            sqlCmd="insert into wma_init (initparam, initvalue) values ('wma_build_id', '$WMA_BUILD_ID'), ('hostname', '$HOSTNAME'), ('is_active', 'true');"
+            _exec_mysql "$sqlCmd" $wmaDBName || return
+            ;;
+        *)
+            echo "$FUNCNAME: ERROR: Unknown or not set Agent Flavor"
+            return $(false)
+            ;;
+    esac
+}
+
+_status_of_couch(){
+    echo "$FUNCNAME:"
+    curl -s $COUCH_HOST:$COUCH_PORT
+    local errVal=$?
+    [[ $errVal -ne 0 ]] && { echo "$FUNCNAME: ERROR: CouchDB database unreachable!"; return $(false) ;}
+    echo "$FUNCNAME: CouchDB connection is OK!"
+}
+
+_status_of_mysql(){
+    echo "$FUNCNAME:"
+    mysqladmin -u $MYSQL_USER --password=$MYSQL_PASS -h 127.0.0.1  status
```

Comment:
Given that it is used in multiple places, I would suggest creating something like:
```
MYSQL_ADDRESS=127.0.0.1
```
at the top of this file and update it throughout this file accordingly.

Reply:
I would go even further and put it in the WMAgent.secrets file, such that it is visible from on every step following `_load_wamgentsecrets` and becomes fully configurable. I'll do that.